DiffusionPreprocessing/scripts/basic_preproc_best_b0.sh

```diff
@@ -0,0 +1,275 @@
+      ${FSLDIR}/bin/bet ${select_b0_dir}/all_b0s_mcf_avg.nii.gz ${select_b0_dir}/nodif_brain -m -R -f 0.3
+      scores=( `${FSLDIR}/bin/fslstats -t ${select_b0_dir}/all_b0s_mcf_res -k ${select_b0_dir}/nodif_brain_mask -M` )
+      scores_str="${scores[@]}"
+
+      # Recomputes the average using only the b0's with scores below (median(score) + 2 * mad(score)),
+      # where mad is the median absolute deviation.
+      idx=`fslpython -c "from numpy import median, where; sc = [float(s) for s in '${scores_str}'.split()]; print(','.join(str(idx) for idx in where(sc < median(abs(sc - median(sc)) * 2 + median(sc)))[0]))"`
+      ${FSLDIR}/bin/fslselectvols -i ${select_b0_dir}/all_b0s_mcf -o ${select_b0_dir}/all_b0s_mcf_avg --vols=${idx} -m
+    done
+  fi
+  echo "b0 scores for ${pe_sign}: " "${scores[@]}"
+
+  printf "%s\n" "${scores[@]}" > ${select_b0_dir}/scores.txt
+done
+
+
+################################################################################################
+## b0 extraction and Creation of acquisition paramater file for topup/eddy
+################################################################################################
+echo "Find the best B0 in the positive and negative volumes"
+
+if [[ -f ${rawdir}/index_best_b0s.txt ]] ; then rm ${rawdir}/index_best_b0s.txt ; fi
+
+for pe_sign in ${basePos} ${baseNeg} ; do
+  # find index of minimum score
+  scores=()
+  while read line; do scores+=("$line"); done < ${rawdir}/select_b0_${pe_sign}/scores.txt
+  min_idx=0
+  for idx in $(seq 0 $((`${FSLDIR}/bin/fslval ${rawdir}/all_${pe_sign}_b0s dim4` - 1))) ; do
+    if [ $(echo "${scores[${idx}]} < ${scores[${min_idx}]}" | bc -l) -eq 1 ] ; then min_idx=$idx ; fi
+  done
+  echo "Selecting ${pe_sign} B0 with index ${min_idx}" 
+  echo "${pe_sign} ${min_idx}" >> ${rawdir}/index_best_b0s.txt
+  ${FSLDIR}/bin/fslroi ${rawdir}/all_${pe_sign}_b0s ${rawdir}/best_${pe_sign}_b0 ${min_idx} 1
+done
+
+# producing acqparams.txt
+if [ ${PEdir} -eq 1 ]; then    #RL/LR phase encoding
+  echo 1 0 0 ${ro_time} > ${rawdir}/acqparams.txt
+  echo -1 0 0 ${ro_time} >> ${rawdir}/acqparams.txt
+elif [ ${PEdir} -eq 2 ]; then  #AP/PA phase encoding
+  echo 0 1 0 ${ro_time} > ${rawdir}/acqparams.txt
+  echo 0 -1 0 ${ro_time} >> ${rawdir}/acqparams.txt
+fi
+
+################################################################################################
+## Merging Files and correct number of slices 
+################################################################################################
+echo "Merging Pos and Neg images"
+${FSLDIR}/bin/fslmerge -t ${rawdir}/Pos `echo ${rawdir}/${basePos}_[0-9]*.nii*`
+${FSLDIR}/bin/fslmerge -t ${rawdir}/Neg `echo ${rawdir}/${baseNeg}_[0-9]*.nii*`
+
+paste -d' ' `echo ${rawdir}/${basePos}*.bval` >${rawdir}/Pos.bval
+paste -d' ' `echo ${rawdir}/${basePos}*.bvec` >${rawdir}/Pos.bvec
+paste -d' ' `echo ${rawdir}/${baseNeg}*.bval` >${rawdir}/Neg.bval
+paste -d' ' `echo ${rawdir}/${baseNeg}*.bvec` >${rawdir}/Neg.bvec
```

Comment:
I see a similar construction was used in the original `basic_preproc.sh`, but given that `${basePos}=Pos`, and `*` matches against no characters, isn't there a danger with this syntax if the script is run on top of already existing results? i.e., if `${rawdir}/Pos.bval` already exists, this will not generate the intended outputs.